tools/pepsi/src/power_off.rs

```diff
@@ -1,29 +1,60 @@
 use clap::Args;
 use color_eyre::{eyre::WrapErr, Result};
 
+use futures_util::{stream::FuturesUnordered, StreamExt};
 use nao::Nao;
 
-use crate::{parsers::NaoAddress, progress_indicator::ProgressIndicator};
+use crate::{
+    parsers::{number_to_ip, Connection, NaoAddress},
+    progress_indicator::ProgressIndicator,
+};
 
 #[derive(Args)]
 pub struct Arguments {
     /// The NAOs to power off e.g. 20w or 10.1.24.22
-    #[arg(required = true)]
-    pub naos: Vec<NaoAddress>,
+    #[arg(long)]
+    pub naos: Option<Vec<NaoAddress>>,
+
+    #[arg(long)]
+    pub all: bool,
 }
 
 pub async fn power_off(arguments: Arguments) -> Result<()> {
-    ProgressIndicator::map_tasks(
-        arguments.naos,
-        "Powering off...",
-        |nao_address, _progress_bar| async move {
-            let nao = Nao::try_new_with_ping(nao_address.ip).await?;
-            nao.power_off()
-                .await
-                .wrap_err_with(|| format!("failed to power {nao_address} off"))
-        },
-    )
-    .await;
+    if arguments.all {
+        let nao_number = 21..=37;
```

Comment:
I don't really know how to solve this, but I don't like the numbers hardcoded here (maybe include_bytes! of hardware_ids.json, then pepsi would still when not in the hulk directory)

Reply:
done